src/router/db.ts

```diff
@@ -46,6 +46,27 @@ db.post('/connect', async (req, res) => {
   }
 });
 
+db.post('/attach', async (req, res) => {
+  console.log('Calling /attach');
+  const {dbAlias, awsRegion, awsAccessKeyId, awsSecretAccessKey} = req.body;
+  if (!dbAlias || !awsRegion || !awsAccessKeyId || !awsSecretAccessKey) return res.status(400).json(
+    `Required key(s) not provided: ${[
+      'awsRegion', 'dbAlias', 'awsAccessKeyId', 'awsSecretAccessKey'
+    ].filter(k => !req.body.hasOwnProperty(k)).join(', ')}`
+  );
+  try {
+    const uid = dbMan.getUid(req.user);
```

Comment:
Wait, how does this work from their AWS lambda function?

Reply:
I was already passing the auth token from the UI to the template.

Reply:
https://github.com/iasql/iasql-engine/pull/587/files#diff-48dd46b95b19450a4573401c27f37a6bbc0a0806e002c71f8572bfd44c0c2ebcR16

Reply:
Got it. Very clever.